omigami/plotting.py

```diff
@@ -0,0 +1,60 @@
+from PIL.PngImagePlugin import PngImageFile
+from pandas import DataFrame
+from rdkit import Chem
+from rdkit.Chem import Draw
+
+
+def _raise_errors(spectra_matches: DataFrame, representation: str = 'smiles'):
+    if representation not in ["smiles", "inchi"]:
+        raise ValueError(
+            f"Got unexpected representation string. Needs to be either 'smiles' or 'inchi' got {representation}"
+        )
+
+    if not isinstance(spectra_matches, type(DataFrame())):
+        raise ValueError(
+            f"Matches need to be a Pandas Dataframe got {type(spectra_matches)}"
+        )
+
+
+def _clean_matches(spectra_matches: DataFrame, representation: str) -> DataFrame:
+    spectra_matches = spectra_matches.drop_duplicates('compound_name')
+    spectra_matches = spectra_matches.drop_duplicates(representation)
+    spectra_matches = spectra_matches[spectra_matches[representation] != ""]
+    spectra_matches = spectra_matches[spectra_matches[representation].notna()]
+
+    return spectra_matches
+
+
+def plot_molecule_structure_grid(spectra_matches: DataFrame, representation: str = 'smiles',
+                                 sort_values: bool = True) -> PngImageFile:
+    """
+    Generate a grid image representation of the hits returned from Spec2Vec and MS2DeepScore outputs.
+    All structures passed MUST have valid smiles or inchi representations.
+
+    Parameters:
+    ----------
+    spectra_matches: DataFrame
+    representation: str = 'smiles' or 'inchi'
+    sort_values: bool = True
+
+    Returns:
+        A Plot showing the structure of the passed smiles/inchis
+    """
+
+    _raise_errors(spectra_matches, representation)
+
+    if sort_values:
+        spectra_matches = spectra_matches.sort_values('score', ascending=True)
```

Comment:
they should be already sorted??  🤔  why did you feel you needed to do this?? did you get not sorted results?? because I think I've seen that happen and I dunno why..

Reply:
They sadly look like this:
![temp](https://user-images.githubusercontent.com/49242091/127126675-f035b541-ed2c-406b-837e-f0186e73ab96.png)

Reply:
huumm not good not good. we need to investigate